core/src/main/java/io/crate/types/CollectionType.java

```diff
@@ -38,12 +38,18 @@
 
     /**
      * Construct a new Collection type
-     * @param innerType May be null if the type is read later through the Streamer interface
+     * @param innerType The type of the elements inside the collection
      */
     public CollectionType(DataType<?> innerType) {
         this.innerType = innerType;
     }
 
+    /**
+     * Constructor used for the Streamer interface which initializes
```

Comment:
Generally this is a problem of the `Streamable` interface but here it is Crate's `Streamer`.